src/Instrumentation/Symfony/src/MessengerInstrumentation.php

```diff
@@ -168,6 +174,69 @@ public static function register(): void
                     $span->setStatus(StatusCode::STATUS_ERROR, $exception->getMessage());
                 }
 
+                $span->end();
+            }
+        );
+
+         // Instrument the receiving of messages (consumer-side)
+         hook(
+            SenderInterface::class,
```

Comment:
no such thing as SenderInterface::receive -> https://github.com/symfony/messenger/blob/7.1/Transport/Sender/SenderInterface.php

Reply:
AFAIK, it should target these interfaces:

https://github.com/symfony/messenger/blob/7.1/MessageBusInterface.php
https://github.com/symfony/messenger/blob/7.1/Transport/Receiver/ReceiverInterface.php

1. `ReceiverInterface::get` is OTEL `receive`
2. `ReceiverInterface::ack` and ``ReceiverInterface::reject` is OTEL `settle` (whichwe don't need to add here)
3. `MessageBusInterface::dispatch` is OTEL `process`

Reply:
See:
https://github.com/symfony/messenger/blob/a391d2cb2b15436c0e3fbdf983440e762a30f88b/Worker.php#L99-L103

and

https://github.com/symfony/messenger/blob/a391d2cb2b15436c0e3fbdf983440e762a30f88b/Worker.php#L162

Reply:
Looking at the example, I guess actually both `ReceiverInterface::get` and `QueueReceiverInterface::getFromQueues` is equivalent to OTEL `receive`

https://github.com/symfony/messenger/blob/7.1/Transport/Receiver/QueueReceiverInterface.php